File: app/src/main/java/com/yao/zhihudaily/ui/BaseActivity.java
package com.yao.zhihudaily.ui;

import android.os.Bundle;

import com.umeng.analytics.MobclickAgent;
import com.yao.zhihudaily.tool.EventCenter;

import androidx.appcompat.app.AppCompatActivity;


/**
 * @author Yao
 * @date 2016/10/1
 */
public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EventCenter.register(this);
    }

    @Override
    public void onResume() {
        super.onResume();
        MobclickAgent.onResume(this);
    }

    @Override
    public void onPause() {
        super.onPause();
        MobclickAgent.onPause(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        EventCenter.unregister(this);
    }
}

File: app/src/main/java/com/yao/zhihudaily/ui/BaseActivity.kt
package com.yao.zhihudaily.ui

import android.os.Bundle

import com.yao.zhihudaily.tool.EventCenter

import androidx.appcompat.app.AppCompatActivity


/**
 * @author Yao
 * @date 2016/10/1
 */
abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        EventCenter.register(this)
    }

    public override fun onResume() {
        super.onResume()
    }

    public override fun onPause() {
        super.onPause()
    }

    override fun onDestroy() {
        super.onDestroy()
        EventCenter.unregister(this)
    }
}
